### PR TITLE
Fix exception tests where we are asserting against the literal message

### PR DESCRIPTION
**Pull request by layomia**

### CODE REVIEW

File: src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs [hunk 1]
```diff
@@ -69,22 +69,27 @@ public void DeserializeBoolean_EmptyString()
...
 
             e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<IDictionary<string, int>>("{\"key\":1,"));
-            Assert.Equal("Expected start of a property name or value, but instead reached end of data. Path: $.key | LineNumber: 0 | BytePositionInLine: 8.", e.Message);
+            Assert.Contains("Path: $.key | LineNumber: 0 | BytePositionInLine: 8.", e.Message);
+            e = null;
 
             e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<IDictionary<string, int>>("{\"key\":1"));
-            Assert.Equal("'1' is an invalid end of a number. Expected a delimiter. Path: $.key | LineNumber: 0 | BytePositionInLine: 8.", e.Message);
+            Assert.Contains("Path: $.key | LineNumber: 0 | BytePositionInLine: 8.", e.Message);
+            e = null;
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs]: Remove the null assignment.

File: src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs [hunk 2]
```diff
@@ -69,22 +69,27 @@ public void DeserializeBoolean_EmptyString()
         public void IncompleteContainers()
         {
             JsonException e = Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<IList<object>>("[1,"));
-            Assert.Equal("Expected start of a property name or value, but instead reached end of data. Path: $[1] | LineNumber: 0 | BytePositionInLine: 2.", e.Message);
+            Assert.Contains("Path: $[1] | LineNumber: 0 | BytePositionInLine: 2.", e.Message);
+            e = null;
```
Comment (stephentoub) [inline src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs]: What's the purpose of all of these `e = null;` lines?  The very next line after each reassigns `e`.

Comment (ahsonkhan) [reply src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs]: jinx ;p

Comment (layomia) [reply src/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs]: Removed :')

**Pull request merged**